packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -513,6 +513,487 @@ export class CloudFormationDeleteStackAction extends CloudFormationDeployAction
+   *
+   * @default No deployment target. This or deploymentTargets is required.
+   */
+  readonly deploymentTargetsPath?: codepipeline.ArtifactPath;
+
+  /**
+   * Note
+   * When this parameter is selected, you must also select DeploymentTargets.
+   *
+   * A list of AWS Regions where stack set instances are created or updated. Regions are updated in the order in which they are entered.
+   *
+   * Enter a list of valid AWS Regions in the format Region1,Region2, as shown in the following example.
+   *
+   * [ "us-west-2", "us-east-1" ]
+   */
+  readonly regions: string | Array<string>;
+
+  /**
+   * The percentage of accounts per Region for which this stack operation can fail before AWS CloudFormation stops the operation in that Region. If
+   * the operation is stopped in a Region, AWS CloudFormation doesn't attempt the operation in subsequent Regions. When calculating the number
+   * of accounts based on the specified percentage, AWS CloudFormation rounds down to the next whole number.
+   *
+   * @default 0
+   */
+  readonly failureTolerancePercentage?: number;
+
+  /**
+   * The maximum percentage of accounts in which to perform this operation at one time. When calculating the number of accounts based on the specified
+   * percentage, AWS CloudFormation rounds down to the next whole number. If rounding down would result in zero, AWS CloudFormation sets the number as
+   * one instead. Although you use this setting to specify the maximum, for large deployments the actual number of accounts acted upon concurrently
+   * may be lower due to service throttling.
+   *
+   * @default 1
+   */
+  readonly maxConcurrentPercentage?: number;
+}
+
+/**
+ * CodePipeline action to deploy a stackset.
+ *
+ * CodePipeline offers the ability to perform AWS CloudFormation StackSets operations as part of your CI/CD process. You use a stack set to create
+ * stacks in AWS accounts across AWS Regions by using a single AWS CloudFormation template. All the resources included in each stack are defined by
+ * the stack set’s AWS CloudFormation template. When you create the stack set, you specify the template to use, as well as any parameters and
+ * capabilities that the template requires.
+ *
+ * For more information about concepts for AWS CloudFormation StackSets, see StackSets concepts in the AWS CloudFormation User Guide.
+ */
+export class CloudFormationStackSetAction extends Action {
+  private static getArtifacts(props: CloudFormationStackSetActionProps) {
+    const artifacts: codepipeline.Artifact[] = [props.templatePath.artifact];
+
+    if (props.parametersPath !== undefined) {
+      artifacts.push(props.parametersPath.artifact);
+    }
+    if (props.deploymentTargetsPath !== undefined) {
+      artifacts.push(props.deploymentTargetsPath.artifact);
+    }
+
+    // de-dupe
+    return artifacts.filter((artifact, index, self) => self.indexOf(artifact) === index);
```

Comment:
No reason to do this (the CodePipeline construct does all of this for you).

This entire method can be safely removed.